test/functional/tests/ci/test_basic.py

```diff
@@ -77,3 +77,70 @@ def test_negative_start_cache():
         )
         if not check_stderr_msg(output, start_cache_with_existing_id):
             TestRun.fail(f"Received unexpected error message: {output.stderr}")
+
+
+@pytest.mark.parametrizex("filesystem", Filesystem)
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_data_integrity(cache_mode, cache_line_size, filesystem):
+    """
+    title: Check basic data integrity after stopping the cache
+    pass criteria:
+        - System does not crash.
+        - All operations complete successfully.
+        - Data consistency is preserved.
+    """
+    cache_id = core_id = 1
+    mountpoint = "/mnt"
+    filepath = f"{mountpoint}/file"
+
+    with TestRun.step("Prepare partitions for cache (2000MiB) and for core (4000MiB)"):
+        cache_device = TestRun.disks["cache"]
+        cache_device.create_partitions([Size(2000, Unit.MebiByte)])
+        cache_part = cache_device.partitions[0]
+
+        core_device = TestRun.disks["core"]
+        core_device.create_partitions([Size(4000, Unit.MebiByte)])
```

Comment:
Limiting core size would reduce time it takes to make a FS. Maybe make it 100MiB?